mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraBlobsDAOTest.java

```diff
@@ -0,0 +1,114 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.mailbox.cassandra.modules.CassandraBlobModule;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+
+import com.datastax.driver.core.utils.UUIDs;
+import com.google.common.base.Strings;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CassandraBlobsDAOTest {
+    private CassandraCluster cassandra;
+    private CassandraBlobsDAO testee;
+
+    @Before
+    public void setUp() throws Exception {
+        cassandra = CassandraCluster.create(new CassandraBlobModule());
+        cassandra.ensureAllTables();
+
+        testee = new CassandraBlobsDAO(cassandra.getConf());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        cassandra.clearAllTables();
+        cassandra.close();
+    }
+
+    @Test
+    public void saveShouldReturnEmptyWhenNullData() throws Exception {
+        Optional<UUID> uuid = testee.save(null).join();
+
+        assertThat(uuid.isPresent()).isFalse();
+    }
+
+    @Test
+    public void saveShouldReturnEmptyWhenEmptyData() throws Exception {
+        Optional<UUID> uuid = testee.save(new byte[]{}).join();
+
+        assertThat(uuid.isPresent()).isFalse();
```

Comment:
I would expect to have an empty byte array as a return here. And thus a UUID. Can you fix it?